### PR TITLE
Updated hdkey dependency from v1.1.1 to v2.0.1

### PR DESCRIPTION
**Pull request by holgerd77**
```shell
$ npm ls secp256k1
ethereumjs-wallet@0.6.3 PATH/ethereumjs-wallet
├─┬ ethereumjs-util@7.0.2
│ └── secp256k1@4.0.1
└─┬ hdkey@2.0.0
  └── secp256k1@4.0.1  deduped
```

Yay.  😄  🎉

### CODE REVIEW

**Comment by ryanio** [comment]
hm that's not good, 3 tests failing ([ci link](https://github.com/ethereumjs/ethereumjs-wallet/pull/127/checks?check_run_id=720855750#step:5:129)):

```
  53 passing (3m)
  3 failing

  1) .derivePath()
       should work with m:

      AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- 'xprv9s21ZrQH143K4KqQx9Zrf1eN8EaPQVFxM2Ast8mdHn7GKiDWzNEyNdduJiQLetNUY7PDEYqnGLkhdzLLUSsrnzEMQBrJoTXMsbyAfkNJveM'
+ 'xprv9s21ZrQH143K4KqQx9Zrf1eN8EaPQVFxM2Ast8mdHn7GKiDWzNEyNdduJhWXToy8MpkGcKjxeFWd8oBSvsz4PCYamxR7TX49pSpp3bmHVAY'
      + expected - actual

      -xprv9s21ZrQH143K4KqQx9Zrf1eN8EaPQVFxM2Ast8mdHn7GKiDWzNEyNdduJiQLetNUY7PDEYqnGLkhdzLLUSsrnzEMQBrJoTXMsbyAfkNJveM
      +xprv9s21ZrQH143K4KqQx9Zrf1eN8EaPQVFxM2Ast8mdHn7GKiDWzNEyNdduJhWXToy8MpkGcKjxeFWd8oBSvsz4PCYamxR7TX49pSpp3bmHVAY
      
      at Context.<anonymous> (test/hdkey.spec.ts:78:12)

  2) .derivePath()
       should work with m/44'/0'/0/1:

      AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- 'xprv9zr2Dxaspc1uXaXjwifQSoiBysxq1Bf987enSQrJgAEYWbVLp9Rahgn8Yd2nXqdhjhNJqKE6D8RiqBpsSvpezEyMZg7ht7ByCmSmmTbsiUf'
+ 'xprvA1ErCzsuXhpB8iDTsbmgpkA2P8ggu97hMZbAXTZCdGYeaUrDhyR8fEw47BNEgLExsWCVzFYuGyeDZJLiFJ9kwBzGojQ6NB718tjVJrVBSrG'
      + expected - actual

      -xprv9zr2Dxaspc1uXaXjwifQSoiBysxq1Bf987enSQrJgAEYWbVLp9Rahgn8Yd2nXqdhjhNJqKE6D8RiqBpsSvpezEyMZg7ht7ByCmSmmTbsiUf
      +xprvA1ErCzsuXhpB8iDTsbmgpkA2P8ggu97hMZbAXTZCdGYeaUrDhyR8fEw47BNEgLExsWCVzFYuGyeDZJLiFJ9kwBzGojQ6NB718tjVJrVBSrG
      
      at Context.<anonymous> (test/hdkey.spec.ts:85:12)

  3) .getWallet()
       should work:

      AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- '0xc97a4a9ed38c8da3fde103e96465bc2542a5e40367df1aefd65b7d84fb0c70e9'
+ '0x26cc9417b89cd77c4acdbe2e3cd286070a015d8e380f9cd1244ae103b7d89d81'
      + expected - actual

      -0xc97a4a9ed38c8da3fde103e96465bc2542a5e40367df1aefd65b7d84fb0c70e9
      +0x26cc9417b89cd77c4acdbe2e3cd286070a015d8e380f9cd1244ae103b7d89d81
      
      at Context.<anonymous> (test/hdkey.spec.ts:94:12)
```

Just a cursory first glance at hdkey's [CHANGELOG.md](https://github.com/cryptocoinjs/hdkey/blob/master/CHANGELOG.md#112--2020-04-16), could it possibly be related to `Use bs58check dependency instead of coinstring` in `1.1.2`?

**Comment by alcuadrado** [comment]
I had **very** similar errors lately. In my case, the problem was that in `secp256k1@4.x` some methods modify the buffers you pass them, while that doesn't happen in `3.x`.

**Comment by alcuadrado** [comment]
I fixed it upstream.

File: package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
     "aes-js": "^3.1.1",
     "bs58check": "^2.1.2",
     "ethereumjs-util": "^7.0.2",
-    "hdkey": "^1.1.1",
+    "hdkey": "^2.0.0",
```
Comment (alcuadrado) [inline package.json:49]: It should be `^2.0.1`, as `2.0.0` was deprecated.

**Comment by ryanio** [comment]
great, I'll merge in

**Comment by alcuadrado** [comment]
I didn't know that GH prevented merges until reviewers that requested changes approve a PR, sorry about that 😅

**Comment by ryanio** [comment]
haha no worries, yeah i think it is a repo setting

**Comment by holgerd77** [comment]
@alcuadrado @ryanio Great guys that you figured this out here, Patricio special thanks for the upstream fix, think these things are always the ones where the ecosystem benefits most at the end! 👍 

this we are now really closing in on the `v1.0.0` release. 😄

**Comment by holgerd77** [comment]
> I had **very** similar errors lately. In my case, the problem was that in `secp256k1@4.x` some methods modify the buffers you pass them, while that doesn't happen in `3.x`.

I will circle in @fanatid here since I wonder if this should this really be the case? And woudn't the root-addressing fix rather be in the `secp256k1` library not modifying the buffer values?

**Comment by fanatid** [comment]
Hi. Which methods modify buffers?

**Comment by holgerd77** [comment]
Likely @alcuadrado can give an answer (or a hint) here, you'll have to wait for Argentinian time zone morning though! 😄

**Comment by fanatid** [comment]
ok, just checked code/readme, probably I'll not need wait morning in Argentinian time zone :D

This 4 functions do in-place operations:
- `privateKeyNegate`
- `privateKeyTweakAdd`
- `privateKeyTweakMul`
- `signatureNormalize`

I changed these functions for 4.x because `bitcoin-core/secp256k1` do in-place. [API.md](https://github.com/cryptocoinjs/secp256k1-node/blob/master/API.md) have note about this. Such functions can be filtered from list by searching `output` and functions which will not match but change data included in list above.

Should I make API.md more clear about this? Should I put some dangerous note to README.md? (Not sure how much people read API.md)

**Comment by holgerd77** [comment]
Thanks for the quick check 😄 , I'll nevertheless let an answer open to @alcuadrado since he normally has some deeper rooted opinion than me on questions like these.

**Comment by alcuadrado** [comment]
I'd modify this part of API.md "In place operations. Some functions doing in place operations" to include a list of all of them.

In general, I'd try to avoid changing the semantics of a function without changing its name and/or type, as people tend to just bump version numbers without much care.

**Comment by alcuadrado** [comment]
> In general, I'd try to avoid changing the semantics of a function without changing its name and/or type, as people tend to just bump version numbers without much care.

This can still be done by deprecating the functions with a warning, and expose them with another name. Something like:

```ts
import * as secp256k1 from  "secp256k1";

export privateKeyTweakAdd(privateKey: Uint8Array, tweak: Uint8Array): Uint8Array {
  console.warn("privateKeyTweakAdd is deprecated. Please use privateKeyTweakAddInPlace");
  return privateKeyTweakAddInPlace(privateKey, tweak);
}

export privateKeyTweakAddInPlace(privateKey: Uint8Array, tweak: Uint8Array): Uint8Array {
  return secp256k1.privateKeyTweakAddInPlace(privateKey, tweak);
}
```

**Comment by holgerd77** [comment]
Yes, I would also agree that it is a good suggestion to deprecate the existing functions, otherwise this can easily lead to serious problems if people do their usual upgrading and don't have a close-enough look.

**Comment by holgerd77** [comment]
Eventually I would even be a bit more explicit on the warning message, like:

> ```ts
> import * as secp256k1 from  "secp256k1";
> 
> export function privateKeyTweakAdd(privateKey: Uint8Array, tweak: Uint8Array): Uint8Array {
>   console.warn("privateKeyTweakAdd() is deprecated. Please use privateKeyTweakAddInPlace().");
>   console.warn("Attention: both versions of the function changed to do operations in-place.");
>   return privateKeyTweakAddInPlace(privateKey, tweak);
> }
> 
> export function privateKeyTweakAddInPlace(privateKey: Uint8Array, tweak: Uint8Array): Uint8Array {
>   return secp256k1.privateKeyTweakAddInPlace(privateKey, tweak);
> }
> ```

**Pull request merged**